lib/ansible/modules/dnf.py

```diff
@@ -1060,6 +1073,11 @@ def ensure(self):
                             results=[],
                         )
 
+            if self.with_optional:
+                package_types = list(dnf.const.GROUP_PACKAGE_TYPES) + ['optional']
+            else:
+                package_types = dnf.const.GROUP_PACKAGE_TYPES
```

Comment:
Should we do `list(dnf.const.GROUP_PACKAGE_TYPES)` here just to make sure `package_types` is of the same type regardless of which branch `if`/`else` is hit? It might prevent errors like `AttributeError` when someone does something to `package_types` somewhere down below in future and assumes it is a list?

Reply:
Good idea. I fixed it, now consistently a list.